bindata/oauth-apiserver/deploy.yaml

```diff
@@ -0,0 +1,147 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: openshift-oauth-apiserver
+  name: apiserver
+  labels:
+    app: openshift-oauth-apiserver
+    apiserver: "true"
+# The number of replicas will be set in code to the number of master nodes.
+spec:
+  selector:
+    matchLabels:
+      app: openshift-oauth-apiserver
+      apiserver: "true"
+  template:
+    metadata:
+      name: openshift-oauth-apiserver
+      labels:
+        app: openshift-oauth-apiserver
+        apiserver: "true"
+    spec:
+      serviceAccountName: oauth-apiserver-sa
+      priorityClassName: system-node-critical
+      initContainers:
+        - name: fix-audit-permissions
+          terminationMessagePolicy: FallbackToLogsOnError
+          image: ${IMAGE}
+          imagePullPolicy: IfNotPresent
+          command: ['sh', '-c', 'chmod 0700 /var/log/oauth-apiserver']
+          securityContext:
+            privileged: true
+          volumeMounts:
+            - mountPath: /var/log/oauth-apiserver
+              name: audit-dir
+      containers:
+      - name: oauth-apiserver
+        terminationMessagePolicy: FallbackToLogsOnError
+        image: ${IMAGE}
+        imagePullPolicy: IfNotPresent
+        command: ["/bin/bash", "-ec"]
+        args:
+          - |
+            if [ -s /var/run/configmaps/trusted-ca-bundle/tls-ca-bundle.pem ]; then
+              echo "Copying system trust bundle"
+              cp -f /var/run/configmaps/trusted-ca-bundle/tls-ca-bundle.pem /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
+            fi
+            exec oauth-apiserver start \
+              --secure-port=8443 \
+              --audit-log-path=/var/log/oauth-apiserver/audit.log \
+              --audit-log-format=json \
+              --audit-log-maxsize=100 \
+              --audit-log-maxbackup=10 \
+              --audit-policy-file=/var/run/configmaps/config/audit-policy.yaml \
+              --etcd-servers=https://etcd.openshift-etcd.svc:2379 \
+              --etcd-cafile=/var/run/configmaps/etcd-serving-ca/ca-bundle.crt \
+              --etcd-keyfile=/var/run/secrets/etcd-client/tls.key \
+              --etcd-certfile=/var/run/secrets/etcd-client/tls.crt \
+              --shutdown-delay-duration=3s \
+              --tls-private-key-file=/var/run/secrets/serving-cert/tls.key \
+              --tls-cert-file=/var/run/secrets/serving-cert/tls.crt \
+              ${FLAGS}
+          # TODO: enable encryption support
+          # --encryption-provider-config=/var/run/secrets/encryption-config
```

Comment:
why did this not stop the openshift-apiserver from launching without enabled encryption? Do we always have an file, but empty then? What is missing to have the same here (follow-up is fine)?

Reply:
`encryption-config` secret is optional, scroll down a few lines. So as it is now no file is being used. I need to add `encryption` controllers and add `revision` support.